js/index.js

```diff
@@ -43,27 +43,82 @@ function renderPepperoni() {
 }
 
 function renderMushrooms() {
-  // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll('.mushroom').forEach(shroom => {
+    if (state.mushrooms) {
+      shroom.style.visibility = 'visible';
+    } else {
+      shroom.style.visibility = 'hidden';
+    }
+  });
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll('.green-pepper').forEach(pepper => {
+    if (state.greenPeppers) {
+      pepper.style.visibility = 'visible';
+    } else {
+      pepper.style.visibility = 'hidden';
+    }
+  });
 }
 
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll('.sauce-white').forEach(x => {
+    if (state.whiteSauce) {
+      x.style.visibility = 'visible';
+    } else {
+      x.style.visibility = 'hidden';
+    }
+});
 }
 
 function renderGlutenFreeCrust() {
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
-}
+  document.querySelectorAll('.crust-gluten-free').forEach(x => {
+  if (state.glutenFreeCrust) {
+    x.style.visibility = 'hidden';
+  } else {
+    x.style.visibility = 'visible';
+  }
+})}
 
 function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
-}
+  document
+  .querySelectorAll(".btn")
+  .forEach((btn, index) => {
+    if (Object.values(state)[index]){
+      btn.classList.add("active")
+    } else if (!Object.values(state)[index]) {
+      btn.classList.remove("active")
+  }});
+};
 
 function renderPrice() {
   // Iteration 4: change the HTML of `<aside class="panel price">`
+  
+  //calculate total
+  let total = basePrice;
+  for (let i in Object.values(state)) {
+    if (Object.values(state)[i]) {
+      total += Object.values(ingredients)[i].price;
+    }
+  }
+  //display total
+  document.getElementById("total").innerHTML = total;
+
+  //show elements on selected
+  let priceI = document.querySelectorAll(".priceI");
+
+  priceI.forEach((item, index) => {
+    if (!Object.values(state)[index]){
+      item.style.visibility = "hidden"
```

Comment:
I would change the visibility here to none, this way there are no empty spaces between prices its looks better :)